mne/io/eeglab/eeglab.py

```diff
@@ -0,0 +1,173 @@
+
+
+class RawSet(_BaseRaw):
+    """Raw object from EEGLAB .set file.
+
+    Parameters
+    ----------
+    fname : str
+        Path to the .set file.
+    ch_fname : str
+        Path to the file containing channel locations. Currently,
+        only .loc extension is supported.
+    eog : list or tuple
+        Names of channels or list of indices that should be designated
+        EOG channels. If None (default), the channel names beginning with
+        ``EOG`` are used.
+    preload : bool or str (default False)
+        Preload data into memory for data manipulation and faster indexing.
+        If True, the data will be preloaded into memory (fast, requires
+        large amount of memory). If preload is a string, preload is the
+        file name of a memory-mapped file which is used to store the data
+        on the hard drive (slower, requires less memory).
+    verbose : bool, str, int, or None
+        If not None, override default verbose level (see mne.verbose).
+
+    Returns
+    -------
+    raw : Instance of RawSet
+        A Raw object containing EEGLAB .set data.
+    """
+    def __init__(self, fname, ch_fname, eog=None, preload=False, verbose=None):
+        """Read EEGLAB .set file.
+        """
+        basedir = op.dirname(fname)
+        eeg = io.loadmat(fname, struct_as_record=False, squeeze_me=True)['EEG']
+
+        # read the data
+        data_fname = op.join(basedir, eeg.data)
+        logger.info('Reading %s' % data_fname)
+
+        last_samps = [int(eeg.xmax * eeg.srate)]
+
+        # get info
+        if ch_fname is not None:
+            ch_fname = op.join(basedir, ch_fname)
+        info = _get_info(eeg, eog, ch_fname)
+
+        super(RawSet, self).__init__(
+            info, preload, filenames=[data_fname], last_samps=last_samps,
+            orig_format='double', verbose=verbose)
+        logger.info('    Range : %d ... %d =  %9.3f ... %9.3f secs'
+                    % (self.first_samp, self.last_samp,
+                       float(self.first_samp) / self.info['sfreq'],
+                       float(self.last_samp) / self.info['sfreq']))
+
+    def _read_segment_file(self, data, idx, fi, start, stop, cals, mult):
+        """Read a chunk of raw data"""
+        scaling = 1e-6
+        nchan = self.info['nchan']
+        data_offset = self.info['nchan'] * start * 4
+        data_left = (stop - start) * nchan
+        # Read up to 100 MB of data at a time.
+        blk_size = min(data_left, (50000000 // nchan) * nchan)
+
+        with open(self._filenames[fi], 'rb', buffering=0) as fid:
+            fid.seek(data_offset)
+            # extract data in chunks
+            for blk_start in np.arange(0, data_left, blk_size) // nchan:
+                blk_size = min(blk_size, data_left - blk_start * nchan)
+                block = np.fromfile(fid,
+                                    dtype=np.float32, count=blk_size) * scaling
+                block = block.reshape(nchan, -1, order='F')
```

Comment:
Numpy documentation says:

""This will be a new view object if possible; otherwise, it will be a copy.""

here: http://docs.scipy.org/doc/numpy-1.10.1/reference/generated/numpy.reshape.html. So, I suppose we're good?